cookbooks/continuous_delivery.rst

```diff
@@ -0,0 +1,187 @@
+   the deployed environments as a commit status or as a comment to the
+   discussion.
+
+Please refer to our integrations list to the complete list of the available
+integrations and their specific configuration.
+
+Custom implementation
+---------------------
+
+For some projects, it might be more appropriate to use a custom implementation.
+With a custom implementation one project can have a semi-automated or a
+completely automated delivery pipeline based on the requirements. In this
+context, the final logic will depend on the project requirements but you will
+find bellow a list of best practices, tips and pointers to help you in your
+implementation.
+
+* `Requirements <cd-requirements>`_
+* `Create and use a machine user for authentication <machine-user>`_
+* `CLI flags and environment variables <cd-env-flags>`_
+* `Creating and deleting environments <cd-env-create-delete>`_
+* `URLs <cd-url>`_
+
+.. _cd-requirements:
+
+Requirements
+^^^^^^^^^^^^
+
+The systement implementing a custom SymfonyCloud deployment automation will have
+several requirements:
+
+#. Have the `Symfony CLI <https://symfony.com/download>`_ installed and runnable
+#. Have Git installed and runnable #. Have `authentication`_ completed
+
+Authentication
+^^^^^^^^^^^^^^
+
+If the automation is expected to be run interactivtely by humans, thoses humans
+must first - locally on their machine - setup Git and run ``symfony login`` as
+well as ``symfony account:ssh:key:add``.
+
+.. _machine-user:
+
+In the implementation is to be run in a non-interactive environment (in a system
+like Jenkins, Travis, GitLab CI or GitHub actions for example), we recommend the
+use of a "Machine user". To do so, head to `SymfonyConnect
+<https://connect.symfony.com>`_ and register a new account.
+
+As this context will not be interactive, interactive authentication will not be
+possible, this is why you have to use an :doc:`API token
+</cookbooks/api_tokens>`, additionnaly, as deployment are based on Git, you will
+have to generate an SSH key, and register it in SymfonyCloud:
+
+.. code-block:: bash
+
+   ssh-keygen -N '' -f symfonycloud-ci.key SYMFONY_TOKEN=xxxx symfony
+   account:ssh:key:add symfonycloud-ci.key
+
+.. caution::
+
+   Don't forget to register and expose the ``SYMFONY_TOKEN`` environment value
+   as well as the generated SSH key in your automation system.
+
+.. _cd-env-flags:
+
+CLI flags and environment variables
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The CLI tool defines several flags and environment flags that allow to change
+it's behavior. As a general rule, for automated context, environment variables
+are preferable but flags can be used ponctually if this is required.
+
+SYMFONY_PROJECT
+"""""""""""""""
+
+Setting this environment value make useless to run ``symfony project:link``. Its
+value should be set to the SymfonyCloud project's id that can be found with
+``symfony project:info`` or in the emails sent by SymfonyCloud.
```

Comment:
```suggestion
When this environment variable is set, it will not be necessary to run ``symfony project:link``. 

Its value should be set to the SymfonyCloud project's id that can be found with
``symfony project:info`` or in the emails sent by SymfonyCloud.
```